pkg/bootstrap/bootstrap_plain_git.go

```diff
@@ -207,6 +207,14 @@ func (b *PlainGitBootstrapper) ReconcileComponents(ctx context.Context, manifest
 		b.logger.Successf("installed components")
 	}
 
+	// Reconcile image pull secret if needed
+	if options.ImagePullSecret != "" && options.RegistryCredential != "" {
```

Comment:
shouldn't this step be done before applying the manifest?

Reply:
No, because there is no namespace to deploy to.